src/bomi/player/mainwindow_m.cpp

```diff
@@ -269,6 +269,19 @@ auto MainWindow::Data::connectMenus() -> void
     connect(play(u"chapter"_q)[u"next"_q], &QAction::triggered, p,
             [seekChapter] () { seekChapter(+1); });
 
+    connect(play[u"state"_q], &QAction::triggered, p, [=] () {
+        QString str;
+        str.append(_MSecToString(e.time()));
+        str.append(u"/"_q);
+        str.append(_MSecToString(e.end()));
+        str.append(u" ("_q);
+        str.append(QString::number(e.time() * 100L / e.end()));
+        str.append(u"%), "_q);
+        str.append(QString::number(e.speed()));
+        str.append(u"x"_q);
+        showMessage(str);
+    });
+
```

Comment:
You can make these codes much simpler using Qt's formatting function QString::arg(). For instacne, `QString str("%1/%2 (%3%) %4x); showMessage(str.arg(_MSecToString(e.time()), ...));`.